### DISCUSSION TITLE
Co-App has to be installed everytime since about a week

### DISCUSSION THREAD

**Discussion by Lalarian** [aclap-dev/video-downloadhelper#201] [Help and Q&A] [answered]
### Your question

Hello!

There's an issue here, since about a week or two:
Every time I want to download some video, VDH claims the Co-app was not installed. It has been working for years!).
I then download and install the current Co-app version ( https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-linux-x86_64.tar.bz ), unpack and install it using './vdhcoapp install'. It then is working fine.
Until the next time I need it - when the issues starts over agan, with VDH claiming the Co-app was not installed.
Up o multiple times a day.
I am using arch Linux and Firefox 123.0.
Thank you!

### Addon & CoApp version and details

Co-App 2.0.19 tarball 64 bit
Arch Linux with current updates
Firefox 123.0
co-app installed for user (not root)

**Comment by paulrouget** [comment]
Oh, that's real strange. Thanks for reporting this issue.

- Does it say that the coapp is outdated in the settings?
- When you do: `./vdhcoapp --info` does it say the version is 2.0.19?
- can you copy/paste the info from here: https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F

**Comment by paulrouget** [comment]
Do you use another browser in parallel of Firefox?

**Reply by Lalarian** [reply]
I have Chrome and Chromium installed, but did not open them since.
> Oh, that's real strange. Thanks for reporting this issue.
> 
>     * Does it say that the coapp is outdated in the settings?
> 
>     * When you do: `./vdhcoapp --info` does it say the version is 2.0.19?
> 
>     * can you copy/paste the info from here: https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F


What path would I have to do this in?
Using locate I could not even find the installed vdhcopp executeable.

Here's the information:
version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/ladmin/Downloads/vdhcoapp-2.0.19/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/ladmin",
  "converterBinary": "/home/ladmin/Downloads/vdhcoapp-2.0.19/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

**Reply by Lalarian** [reply, marked as answer]
OK, seeing the information you asked for seems to possibly explain the issue:
Does running ".vdhcoapp install" setup the app in the current path? (Which happens to be the Download folder in my case, which I periodically empty as it's not supposed to be a permanent storag)
I expected the installer to copy the tool to my profile folder (like ~/.vdhcoapp) but not 'Downloads'.

PS:
OK, I just manually created ~/.vdhcoapp, moved the extraced files there and installed from within ¨/.vdhcoapp .
VDH's information now reads:

`version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/ladmin/.vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/ladmin",
  "converterBinary": "/home/ladmin/.vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}
`

Seems like this should already fix the issue. So there's no problem with the code. I think it might be a good idea to add a short comment to the 'CoApp Installation ' notification, that Linux users besides those using the .deb should be doing so manually. 

Thank you! :)
